src/main/java/org/testng/internal/annotations/ConfigurationAnnotation.java

```diff
@@ -27,6 +27,8 @@
   private boolean m_beforeSuite = false;
   private boolean m_afterSuite = false;
   private boolean m_alwaysRun = false;
+  private boolean beforeGroups = false;
```

Comment:
Doc is missing: what is the difference between `beforeGroups` and `m_beforeGroups`?